src/compiler/core.ts

```diff
@@ -1101,6 +1101,19 @@ namespace ts {
         return result;
     }
 
+    /**
+     * Creates a set from the elements of an array.
+     *
+     * @param array the array of input elements.
+     *
+     * This function makes no effort to avoid collisions; if any two elements produce

```

Comment:
This is a `Map<true>`, so the elements of the array aren't present in the result. So `the element with the higher index in the array will be the one associated with the produced key` can't be true.